packages/react-native/ReactAndroid/src/main/jni/CMakeLists.txt

```diff
@@ -8,6 +8,11 @@ set(CMAKE_VERBOSE_MAKEFILE on)
 
 project(ReactAndroid)
 
+# Convert input paths to CMake format (with forward slashes)
+file(TO_CMAKE_PATH "${REACT_ANDROID_DIR}" REACT_ANDROID_DIR)
+file(TO_CMAKE_PATH "${REACT_BUILD_DIR}" REACT_BUILD_DIR)
+file(TO_CMAKE_PATH "${REACT_COMMON_DIR}" REACT_COMMON_DIR)
```

Comment:
@alespergl I'm not an expert of CMake, but can we wrap this change in a [platform check](https://www.openguru.com/2009/04/cmake-detecting-platformoperating.html)?
Something like:
```
if(${CMAKE_SYSTEM_NAME} MATCHES "Win32") 
  file(TO_CMAKE_PATH "${REACT_ANDROID_DIR}" REACT_ANDROID_DIR)
	file(TO_CMAKE_PATH "${REACT_BUILD_DIR}" REACT_BUILD_DIR)
	file(TO_CMAKE_PATH "${REACT_COMMON_DIR}" REACT_COMMON_DIR)
endif
```

I'm scared this could cause harm on other systems, otherwise.

Reply:
using `file(TO_CMAKE_PATH` should actually be correct on every platform.
As the CI is green, we can just merge it 👍

Reply:
Yes, this operation should be cross-platform.

Note: There is a similar function [cmake_path](https://cmake.org/cmake/help/latest/command/cmake_path.html#native-conversion) which seems to be preferred, but requires CMake 3.20 or later, whereas the minimum specified in this `CMakeLists.txt` is 3.13.